lib/dealership.rb

```diff
@@ -0,0 +1,54 @@
+class Dealership
+  attr_reader :inventory
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+    @inventory_count = 0
+    @has_inventory = false
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def inventory_count
+    @inventory.size
+  end
+
+  def has_inventory?
+    @has_inventory
+  end
+
+  def cars_by_make(make)
+    @inventory.select do |car|
+      car.make == make
+    end
+  end
+
+  def total_value
+    total_v = []
+    @inventory.each do |car|
+      total_v << car.total_cost
+    end
+    total_v.reduce(0, :+)
+  end
+
+  def details
+    details = {}
```

Comment:
So very close here. A couple things:
1. You want your keys to be strings, which you are doing with the `.to_s` but you could just do `details["total_value"]`
2. You are what we call `hard coding` the values. So they pass based on the test you wrote but we want it to be dynamic. Is there a method for `total_value`? And do you have your `@address` in the initialize? That is what you would set you keys equal to. 
```
details["total_value"] = total_value
details["address"] = @address
```